eclipse-external-annotations-m2e-plugin.core/src/org/lastnpe/m2e/core/configurator/ClasspathConfigurator.java

```diff
@@ -169,7 +169,7 @@ private void setExternalAnnotationsPath(final IClasspathEntryDescriptor cpEntry,
             return readFile(file.toPath());
         } else if (fileOrDirectory.isFile()) {
             final Path jarFilePath = Paths.get(fileOrDirectory.toURI());
-            final URI jarEntryURI = URI.create("jar:file:" + jarFilePath.toUri().getPath() + "!/" + fileName);
+            final URI jarEntryURI = URI.create("jar:file:" + jarFilePath.toUri().getPath().replace(" ", "%20") + "!/" + fileName);
```

Comment:
this code (which I  must have written...) is bad anyway, and makes the classic mistake of String conversion when there should be a better way. Would you have a minute to try if e.g. this would work instead:

    import org.eclipse.core.runtime.URIUtil;

    jarEntryURI = URIUtil.toJarURI(jarFilePath.toUri(), new Path(fileName));

Reply:
Yes.
```
final URI jarEntryURI = URIUtil.toJarURI(jarFilePath.toUri(), new org.eclipse.core.runtime.Path(fileName));
```
eclipse's Path need FQCN though, because there's a nio Path import in scope already.

Is it ok?